internal/characters/travelerdendro/burst.go

```diff
@@ -0,0 +1,82 @@
+package travelerdendro
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+)
+
+var burstFrames [][]int
+
+const burstHitmark = 91
+const leaLotusAppear = 54
+
+func init() {
+	burstFrames = make([][]int, 2)
+
+	// Male
+	burstFrames[0] = frames.InitAbilSlice(58)
+	burstFrames[0][action.ActionSwap] = 57 // Q -> Swap
+
+	// Female
+	burstFrames[1] = frames.InitAbilSlice(58)
+	burstFrames[1][action.ActionSwap] = 57 // Q -> Swap
+}
+
+func (c *char) Burst(p map[string]int) action.ActionInfo {
+
+	c.SetCD(action.ActionBurst, 1200)
+	c.ConsumeEnergy(2)
+
+	// Duration counts from first hitmark
+	burstDur := burstHitmark - leaLotusAppear + 12*60
+	// burstDur := 12 * 60
+	if c.Base.Cons >= 2 {
+		burstDur += 3 * 60
+	}
+
+	s := c.newLeaLotusLamp(burstDur)
+
+	c.Core.Tasks.Add(func() {
+		c.Core.Combat.AddGadget(s)
+	}, leaLotusAppear)
+
+	c.burstOverflowingLotuslight = 0
+	// Expiry frame for delay conditional
+	var burstExp = c.Core.F + burstDur
+
+	c.Core.Log.NewEvent(fmt.Sprintf("delay start: %s", strconv.Itoa(c.Core.F+leaLotusAppear)), glog.LogCharacterEvent, c.Index)
+	c.Core.Log.NewEvent(fmt.Sprintf("burst expiry: %s", strconv.Itoa(burstExp+leaLotusAppear)), glog.LogCharacterEvent, c.Index)
+
+	delayTick := leaLotusAppear
```

Comment:
I added delayTick because delay is set to the c.Core.F, so on the second loop at from 1200+ you'll be setting a delay of more than 20s, meaning the a1/a6 buffs won't be applied until the rotation after that, with it desyncing further each time.

With delayTick as a separate variable, the start of the buffing is queued to begin when the lotus appears and won't desync.

Reply:
I don't get it, delay is getting reset for each individual loop anyways so there shouldn't be any issues with that

Reply:
Ok, then change the logic so that delay starts at leaLotusAppear and ends at s.Gadget.Duration. Then you won't need delayTick at all